### ISSUE TITLE
Display child timeline on parent object's timeline

### ISSUE THREAD

**Issue by FelixMalfait** [twentyhq/twenty#5066]
### What
It's annoying that when I take a note on a _person_ I don't see it on the _company_ or the _opportunity_.
For workspace that want to do so, we should let them cascade the info automatically into parent timeline.

### Backend
- Introduce a qualifier on relationships "isTimelineInheritanceEnabled"
- Create a custom resolver to retrieve TimelineActivities, in that resolver you will query the metadata schema to find the object's direct children, fetch their timelines, and then merge all timelines together. Ideally we would have cascading depth = 0, 1 or 2 as a parameter to this resolver (along with the object name / id).

### Frontend
- MUST: Adapt the text of TimelineActivities to reflect when activities comes from a children
- NICE TO HAVE: Add a toggle in relationships to enable inheritance (should we say cascading?)
- FOLLOW UP: Later we'll add user-facing filters on the timeline, not part of this issue

**Comment by FelixMalfait** [comment]
cc @Bonapara if you work on design for this, would you mind adding a link there? Not sure if user-facing filters will be tackled as part of this issue or as a second step - but they shouldn't be too long to add. 
Note that the API would allow cascading depth = 0,1,2 but maybe we just want to offer 0,2 in the UI to avoid adding too much complexity...

**Comment by Bonapara** [comment]
@FelixMalfait I was thinking of this for a v1: 

![CleanShot 2024-04-22 at 15 47 04](https://github.com/twentyhq/twenty/assets/19412894/766964e2-4f6a-4e30-8b66-8696e7299903)

We could consider adding the advanced cascading depth settings in the `Object Settings page` later. WDYT?

**Comment by FelixMalfait** [comment]
@Bonapara yes perfect for a toggle on the settings page! I really like that you used the object name to make it explicit, much better than what I would have imagined.
Small detail, we call them activities now I think, so copy could be: "Show activities from the People timeline on the Warehouse timeline"

I was rather referring to the filters on the timeline (in the app) for the depth, defining it at the relationship level would not really be possible. We can live without them for v1 for sure!

**Comment by Bonapara** [comment]
For the copy would "Show People's timeline activities on the Warehouse timeline." also suits? (The shorter the best for the UX)

Ok didn't get you were talking about the filters!

I was thinking/talking of another subject for the future:

"Tomorrow, if I have `Object A`, `Object B`, and `Object C`, and I choose to display `Object C`'s timeline activities on `Object B`'s timeline, can I specify whether `Object C`'s timeline activities should also be displayed on `Object A`'s timeline when setting the `Object A` <> `Object B` relationship? I thought this advanced setting could be introduced later in the object settings."

**Comment by FelixMalfait** [comment]
@Bonapara relation depth could be defined in settings eventually yes you're right, could be inheritanceDepth or a field like that. 

I prefer the first copy without the possessive form (companies' vs people's etc = make it harder to implement in code ; and from/to made the destination more clear imo). But if you prefer, we can come back to your initial version! Let's just change display-> show and events->activities - forget the "from/to" I added

**Comment by Bonapara** [comment]
It would be great if it could fit in one line most of the time. Do you see a shorter version?

![image](https://github.com/twentyhq/twenty/assets/19412894/5e4a8a17-eea0-4456-ad38-8cce43d126bc)

**Comment by FelixMalfait** [comment]
Yes great!
